mhkit/wave/io/cdip.py

```diff
@@ -0,0 +1,461 @@
+    end_date: string 
+        End date in YYYY-MM-DD, e.g. '2012-04-30'        
+        
+    Returns
+    -------
+    start_stamp: float
+         seconds since the Epoch to start_date    
+    end_stamp: float
+         seconds since the Epoch to end_date
+    '''
+    assert isinstance(start_date, (str, type(None))), ('start_date' /
+        'must be of type str')
+    assert isinstance(end_date, (str, type(None))), ('end_date must be' / 
+        'of type str')
+        
+    time_all = nc.variables['waveTime'][:].compressed()
+    time_range_all = [datetime.datetime.fromtimestamp(time_all[0]).replace(tzinfo=timezone.utc), 
+                      datetime.datetime.fromtimestamp(time_all[-1]).replace(tzinfo=timezone.utc)]
+    
+    if start_date:        
+        start_datetime = _validate_date(start_date)   
+    if end_date:
+        end_datetime = _validate_date(end_date)   
+        if start_datetime > end_datetime:
+            raise Exception(f'start_date ({start_datetime}) must be'+
+                f'before end_date ({end_datetime})')
+        elif start_datetime == end_datetime:
+            raise Exception(f'start_date ({start_datetime}) cannot be'+
+                f'the same as end_date ({end_datetime})')
+    
+    if start_date:
+        if start_datetime > time_range_all[0] and start_datetime < time_range_all[1]:
+            start_stamp = start_datetime.replace(tzinfo=timezone.utc).timestamp()
+        else:
+            print(f'WARNING: Provided start_date ({start_datetime}) is ' 
+            f'not in the returned data range {time_range_all} \n' 
+            f'Setting start_date to the earliest date in range '
+            f'{time_range_all[0]}')
+            start_stamp = pd.to_datetime(time_range_all[0]).replace(tzinfo=timezone.utc).timestamp()         
+    
+    if end_date:
+        if end_datetime > time_range_all[0] and end_datetime < time_range_all[1]:
+            end_stamp = end_datetime.replace(tzinfo=timezone.utc).timestamp()
+        else:
+            print(f'WARNING: Provided end_date ({end_datetime}) is ' 
+            f'not in the returned data range {time_range_all} \n' 
+            f'Setting end_date to the latest date in range '
+            f'{time_range_all[1]}')
+            end_stamp = pd.to_datetime(time_range_all[1]).replace(tzinfo=timezone.utc).timestamp()        
+    
+    
+    if start_date and not end_date:
+        end_stamp = pd.to_datetime(time_range_all[1]).replace(tzinfo=timezone.utc).timestamp()            
+
+    elif end_date and not start_date:
+        start_stamp = pd.to_datetime(time_range_all[0]).replace(tzinfo=timezone.utc).timestamp()
+        
+    if not start_date:
+        start_stamp = pd.to_datetime(time_range_all[0]).replace(tzinfo=timezone.utc).timestamp()
+    if not end_date:
+        end_stamp = pd.to_datetime(time_range_all[1]).replace(tzinfo=timezone.utc).timestamp()
+
+    return start_stamp, end_stamp 
+
+    
+def request_netCDF(station_number, data_type):
```

Comment:
Requesting the NetCDF file is broken out to provide users flexibility by taking a modular approach to the data request. This is useful for the user to make custom workflows beyond what parse_data does currently and can also be used if the user prefers to request data and use a tool such as xarray instead of the returned dictionary of dataframes.

Reply:
My suggestion is to more clearly spell out in the text description in the example notebook examples of reasons why a user would use the broken out functions rather than the parse_data function. You just mentioned a couple of examples and anything else you could add would be great. As the notebook is now it is not completely clear what added value breaking out the workflow would provide and my fear is that it would lead to choice paralysis for the users. Let's spell it out very plainly for the users.

Reply:
The top of the example directs any user immediately to what we believe is the most common function they would want to use. I believe this would prevent any choice paralysis. I have added the text above to the `request_data` function to help address your concern. 

70df9ab